app/channel.go

```diff
@@ -49,12 +49,19 @@ func (a *App) JoinDefaultChannels(teamId string, user *model.User, channelRole s
 	} else {
 		townSquare := result.Data.(*model.Channel)
 
-		cm := &model.ChannelMember{ChannelId: townSquare.Id, UserId: user.Id,
-			Roles: channelRole, NotifyProps: model.GetDefaultChannelNotifyProps()}
+		cm := &model.ChannelMember{
+			ChannelId:   townSquare.Id,
+			UserId:      user.Id,
+			Roles:       channelRole,
+			NotifyProps: model.GetDefaultChannelNotifyProps(),
+		}
 
 		if cmResult := <-a.Srv.Store.Channel().SaveMember(cm); cmResult.Err != nil {
 			err = cmResult.Err
 		}
+		if result := <-a.Srv.Store.ChannelMemberHistory().LogJoinEvent(user.Id, townSquare.Id, model.GetMillis()); result.Err != nil {
+			l4g.Warn("Failed to update ChannelMemberHistory table %v", result.Err)
+		}
```

Comment:
I would suggest a transaction to ensure that the history is updated if and only if the join is successful, but... this seems par for the course I guess. Maybe this would make a decent example for the agenda item / dev talk I'm considering.

Reply:
I'm going to be making the write to ChannelMemberHistory conditional on whether or not the Message Export feature is enabled as a part of [PLT-8238](https://mattermost.atlassian.net/browse/PLT-8238), so I'll look at making the two writes transactional at that time.